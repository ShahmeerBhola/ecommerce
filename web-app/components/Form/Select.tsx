import React from 'react';

type Props = {
    name: string;
    value: string | string[];
    placeholder?: string;
    disabled?: boolean;
    borderColor?: 'border-red-200' | 'border-gray-901';
    onChange?: Forms.InputOnChange;
    onBlur?: Forms.InputOnBlur;
    options: Forms.FieldOption[];
};

const Select: React.FunctionComponent<Props> = ({
    name,
    value,
    placeholder,
    disabled,
    borderColor = 'border-gray-901',
    onChange,
    onBlur,
    options,
}) => {
    return (
        <select
            name={name}
            className={`text-sm text-gray-902 placeholder-gray-902 p-3 focus:outline-none w-full h-50px bg-gray-50 border border-solid ${borderColor}`}
            value={value || placeholder}
            disabled={disabled}
            onChange={onChange}
            onBlur={onBlur}
        >
            <option value={placeholder} disabled>
                {placeholder}
            </option>
            {options.map(({ value: _value, label }) => (
                <option value={_value} key={_value}>
                    {label}
                </option>
            ))}
        </select>
    );
};

export default Select;
