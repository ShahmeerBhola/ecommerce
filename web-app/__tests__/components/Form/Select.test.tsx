import React from 'react';
import renderer from 'react-test-renderer';
import Select from '../../../components/Form/Select';

describe('Select component', () => {
  const options = [
    {
      value: 'foo',
      label: 'Foo',
    },
    {
      value: 'bar',
      label: 'Bar',
    },
  ];

  test('it renders properly when value is empty string', () => {
    const tree = renderer.create(<Select name="my-select" value="" options={options} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders properly when value is non-empty string', () => {
    const tree = renderer.create(<Select name="my-select" value="foo" options={options} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders properly the placeholder', () => {
    const tree = renderer
      .create(<Select name="my-select" value="" placeholder="My Placeholder" options={options} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders properly when disabled is true', () => {
    const tree = renderer.create(<Select name="my-select" value="" disabled options={options} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders properly when borderColor is non-default value', () => {
    const tree = renderer
      .create(<Select name="my-select" value="" borderColor="border-red-200" options={options} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders properly when onChange and onBlur props provided', () => {
    const tree = renderer
      .create(
        <Select
          name="my-select"
          value=""
          onChange={(): void => {
            console.log('onChange');
          }}
          onBlur={(): void => {
            console.log('onBlur');
          }}
          options={options}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
