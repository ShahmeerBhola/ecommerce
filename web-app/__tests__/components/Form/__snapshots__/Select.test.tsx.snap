// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select component it renders properly the placeholder 1`] = `
<select
  className="text-sm text-gray-200 p-3 focus:outline-none w-full h-50px bg-gray-50 border border-solid border-gray-100"
  name="my-select"
  value="My Placeholder"
>
  <option
    disabled={true}
    value="My Placeholder"
  >
    My Placeholder
  </option>
  <option
    value="foo"
  >
    Foo
  </option>
  <option
    value="bar"
  >
    Bar
  </option>
</select>
`;

exports[`Select component it renders properly when borderColor is non-default value 1`] = `
<select
  className="text-sm text-gray-200 p-3 focus:outline-none w-full h-50px bg-gray-50 border border-solid border-red-200"
  name="my-select"
>
  <option
    disabled={true}
  />
  <option
    value="foo"
  >
    Foo
  </option>
  <option
    value="bar"
  >
    Bar
  </option>
</select>
`;

exports[`Select component it renders properly when disabled is true 1`] = `
<select
  className="text-sm text-gray-200 p-3 focus:outline-none w-full h-50px bg-gray-50 border border-solid border-gray-100"
  disabled={true}
  name="my-select"
>
  <option
    disabled={true}
  />
  <option
    value="foo"
  >
    Foo
  </option>
  <option
    value="bar"
  >
    Bar
  </option>
</select>
`;

exports[`Select component it renders properly when onChange and onBlur props provided 1`] = `
<select
  className="text-sm text-gray-200 p-3 focus:outline-none w-full h-50px bg-gray-50 border border-solid border-gray-100"
  name="my-select"
  onBlur={[Function]}
  onChange={[Function]}
>
  <option
    disabled={true}
  />
  <option
    value="foo"
  >
    Foo
  </option>
  <option
    value="bar"
  >
    Bar
  </option>
</select>
`;

exports[`Select component it renders properly when value is empty string 1`] = `
<select
  className="text-sm text-gray-200 p-3 focus:outline-none w-full h-50px bg-gray-50 border border-solid border-gray-100"
  name="my-select"
>
  <option
    disabled={true}
  />
  <option
    value="foo"
  >
    Foo
  </option>
  <option
    value="bar"
  >
    Bar
  </option>
</select>
`;

exports[`Select component it renders properly when value is non-empty string 1`] = `
<select
  className="text-sm text-gray-200 p-3 focus:outline-none w-full h-50px bg-gray-50 border border-solid border-gray-100"
  name="my-select"
  value="foo"
>
  <option
    disabled={true}
  />
  <option
    value="foo"
  >
    Foo
  </option>
  <option
    value="bar"
  >
    Bar
  </option>
</select>
`;
